src/scheduler/pbs_sched_utils.cpp

```diff
@@ -684,13 +744,56 @@ sched_svr_init(void)
  * @return void
  */
 static void
-reconnect_servers()
+reconnect_servers(void)
 {
+	svr_conn_t **prim_conn = get_conn_svr_instances(clust_primary_sock);
+	svr_conn_t **sec_conn = get_conn_svr_instances(clust_secondary_sock);
+	int num_down = 0;
+	static time_t last_reconnect = -1;
+	const int min_reconnect_period = 30;	/* Reconnect at least 30secs after last */
+
 	pthread_mutex_lock(&cleanup_lock);
+	if (prim_conn == NULL || sec_conn == NULL || get_num_servers() == 1) {
+		close_servers();
+		connect_svrpool();
+		pthread_mutex_unlock(&cleanup_lock);
 
-	close_servers();
-	connect_svrpool();
+		return;
+	}
+
+	part_tolerance = true;
```

Comment:
Shouldn't we be setting partition_tolerance after we have checked min_reconnect_period? Can it happen that reconnect_servers is called frequently twice and it makes the scheduler go into partition tolerance mode even when no server was down?

Reply:
reconnect_servers() should only be called when one/more servers is down, so I wanted to set part_tolerance upfront .. but I'll move it to the if condition that checks how many are down ..